shared/src/commonMain/kotlin/com.jetbrains.spacetutorial/SpaceXSDK.kt

```diff
@@ -0,0 +1,22 @@
+package com.jetbrains.spacetutorial
+
+import com.jetbrains.spacetutorial.cache.Database
+import com.jetbrains.spacetutorial.cache.DatabaseDriverFactory
+import com.jetbrains.spacetutorial.entity.RocketLaunch
+import com.jetbrains.spacetutorial.network.SpaceXApi
+
+class SpaceXSDK(databaseDriverFactory: DatabaseDriverFactory, val api: SpaceXApi) {
+    private val database = Database(databaseDriverFactory)
+
+    suspend fun getLaunches(forceReload: Boolean): List<RocketLaunch> {
+        val cachedLaunches = database.getAllLaunches()
+        return if (cachedLaunches.isNotEmpty() && !forceReload) {
+            cachedLaunches
+        } else {
+            api.getAllLaunches().also {
+                database.clearDatabase()
+                database.createLaunches(it)
```

Comment:
I would create a method in the `Database` class that does this, that way the deleting/adding could be done within the same transaction.